### ISSUE TITLE
Template overwriting does not work in Symfony4

### ISSUE THREAD

**Issue by Sengorius** [FriendsOfSymfony/FOSOAuthServerBundle#557]
Hey guys,

I'm now far enough and got everything working with this bundle in Symfony4. The only thing remaining are some improvements in templates, so I followed the [template guide](https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/master/Resources/doc/extending_the_authorization_page.md) with addition of the expected way to do it in Symfony4 (explained [here](https://symfony.com/doc/current/templating/overriding.html)).

Well, maybe I'm just blind, but all of my tries did no provide any result. The template is still the one from the bundle. Here is what I did:

![template_tree_1](https://user-images.githubusercontent.com/7498666/38366168-39702a8e-38df-11e8-91e1-25e3f5e9c09a.png)
![template_tree_2](https://user-images.githubusercontent.com/7498666/38366184-413437c4-38df-11e8-87af-01b7948e38b2.png)
![template_tree_3](https://user-images.githubusercontent.com/7498666/38366191-470670ae-38df-11e8-8178-2c5c43768c11.png)

Of course, I cleared the cache several times. Any ideas?

**Comment by dkarlovi** [comment]
The first one should work. We had https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/pull/519 which fixed the override for Symfony 3.4+

**Comment by Sengorius** [comment]
@dkarlovi Yes, it should, but unfortunately it did not. Maybe it's because you use the templating component and not the twig component as templating endpoint directly?

**Comment by fxbt** [comment]
@Sengorius I had the same problem. The solution for me was to follow the old templating guidelines and put my twig template in `src/Resources/FOSOAuthServerBundle/views/`.

This is not compliant with the Symfony 4 documentation but at least it works.

![image](https://user-images.githubusercontent.com/6928162/38558553-c8da06f0-3cd0-11e8-92f9-b9f38b8f4022.png)

**Comment by Sengorius** [comment]
Thanks @fxbt, this really did work! :+1: 

As I'm about to overwrite more than just the `authorize.html.twig`, I also needed to add the new path to Twig resources:
``` yaml
# config/packages/twig.yaml
twig:
    debug: '%kernel.debug%'
    strict_variables: '%kernel.debug%'
    paths:
        - '%kernel.project_dir%/templates'
        - '%kernel.project_dir%/src/Resources'
```
And change the extend/include in templates:
``` twig
{# src/Resources/FOSOAuthServerBundle/views/Authorize/authorize.html.twig #}
{% extends 'FOSOAuthServerBundle/views/layout.html.twig' %}

{% block fos_oauth_server_content %}
    {% include 'FOSOAuthServerBundle/views/Authorize/authorize_content.html.twig' %}
{% endblock %}
```

Thanks guys. I think we can close this one.

**Comment by dkarlovi** [comment]
> I think we can close this one.

Let's leave it, the bundle should be Symfony4 compatible out of the box. The solution you described is a workaround.

**Comment by dkarlovi** [comment]
I'm currently doing the same thing for my app, it also does NOT work for me, but I have overridden [a template for API Platform](https://github.com/api-platform/core/blob/master/src/Bridge/Symfony/Bundle/Resources/views/SwaggerUi/index.html.twig) and that one overrides OK.

This bundle does exactly the same thing (the way I see it) and so it should work as-is, but it doesn't. I'm confused. :(

**Comment by karser** [comment]
Looks like  `'FOSOAuthServerBundle:Authorize:authorize.html.'` has to be replaced with `'@FOSOAuthServer/Authorize/authorize.html.'` in [AuthorizeController](https://github.com/FriendsOfSymfony/FOSOAuthServerBundle/blob/1.6.1/Controller/AuthorizeController.php#L178) as of [symfony 4.1](https://symfony.com/doc/4.1/templating.html#referencing-templates-in-a-bundle) documentation

**Comment by arnofo** [comment]
Hey guys,

I can tell the change mentionned by @karser has been merged on the master branch for quite some time now. (#564 )

When can we excpect a new release ?

Thanks !

**Comment by tomazartack** [comment]
Just wanted to say, after having issues overriding the template as well, this worked for me on Symfony 4.4:

{project_root}/templates/bundles/FOSOauthServerBundle/Authorize/authorize.html.twig

Content: 
```
{% extends "@FOSOAuthServer/layout.html.twig" %}

{% block fos_oauth_server_content %}
{% include "@FOSOAuthServer/Authorize/authorize_content.html.twig" %}
{% endblock fos_oauth_server_content %}
```

No twig configuration needed. Because I was doing an upgrade from Symfony 3 to Symfony 4, I had to pay attention to the include template syntax.
